File: foody-master/Back End/src/main/java/com/food/service/FoodService.java
package com.food.service;

import java.util.List;

import com.food.model.Food;

public interface FoodService {
	public Food addFood(Food food);
	public List<Food> getAllFood();
	public Food updateFood(Food food);
	public Food getFoodById(int id);
	public void removeFood(int id);
	public List<Food> search(String name);
	public List<Food> sortByPrice();
	public List<Food> sortByPriceDsc();
	public List<Food> sortByName();
	public List<Food> sortByNameDsc();
}

File: foody-master/Front End/src/app/services/food.service.ts
import { Injectable } from '@angular/core';
import { foodUrl } from '../config/api';
import { HttpClient } from '@angular/common/http';
import { Food } from '../model/food';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class FoodService {

  constructor(private http: HttpClient) { }

  updateFood(food: Food){
    return this.http.put<Food>(foodUrl, food);
  }

  getAllFood(): Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl);
  }

  search(name: String):Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl+"/search/"+name);
  }

  sortByPrice():Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl+"/sort/price");
  }

  sortByPriceDsc():Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl+"/sort/price/dsc");
  }

  sortByName():Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl+"/sort/name");
  }

  sortByNameDsc(): Observable<Food[]>{
    return this.http.get<Food[]>(foodUrl+"/sort/name/dsc")
  }

}
